### ISSUE TITLE
Question: What contrib modules do we want to enable by default?

### ISSUE THREAD

**Issue by stephen-cox** [localgovdrupal/localgov#25]
Currently the only contrib module that is enabled at install time is Admin Toolbar.
Devel is being downloaded but not enabled. Do we want to enable it?
Are there any other modules we want to download and/or enable with the Drupal site install?

**Comment by andybroomfield** [comment]
On the BHCC site, devel is enabled on local only installs via a local config split.
Will Devel be a main or dev only dependency?

**Comment by stephen-cox** [comment]
My inclination is, for now, to enable Devel on a site install as it will be useful during development. This would make it a main dependency for now, as otherwise we'll need to write code to check whether it's installed or not. I understand we don't want it as part of a production build, but we're a long way from that at the moment.

Happy to discuss further.

**Comment by andybroomfield** [comment]
@stephen-cox Fair enough. We will need to decide in future if the default install is for a development installation and a special config split for production, or the other way round (BHCC defaults to a production environment unless development splits / config is in use).

**Comment by Adnan-cds** [comment]
Hi Stephen, you have already added the admin_toolbar module.  Thanks for that.  We also have the [adminimal_admin_toolbar](http://drupal.org/project/adminimal_admin_toolbar) in our site which depends on admin_toolbar.  This adminimal_admin_toolbar makes things a little easier on the eyes.  I am saying this either because I have got too used to it Or that is how the rest of humanity feels; I don't know :)  I will leave this to you. @stephen-cox

**Comment by Adnan-cds** [comment]
Hi Stephen,
I have always found the [diff](https://www.drupal.org/project/diff) module useful.

We also use [Elbow room](https://www.drupal.org/project/elbow_room) which is a contrib module originally written for the Brighton site and then contributed back to the community.  We are totally addicted to it.  I was even asked to extend its behaviour which forced me to write some custom Javascript.  That's how popular it has become here.

At Croydon, we are slowing becoming heavy users of the [EntityQueue](https://www.drupal.org/project/entityqueue) module.  That's something to consider.

Of late, we are trying out the [Field formatter class](https://www.drupal.org/project/field_formatter_class) module which lets you add classes to field markup without creating yet another Twig template.  In the past I have found it useful in other projects as well.  This one merits your consideration IMHO.  Those who like *Field formatter class* may also like [Block class](https://www.drupal.org/project/block_class).

I can't imagine life without stage_file_proxy.  But localgov doesn't represent any production site, so this module may not be relevant for this particular code base.

HTH

**Comment by andybroomfield** [comment]
> We also use [Elbow room](https://www.drupal.org/project/elbow_room) which is a contrib module originally written for the Brighton site and then contributed back to the community. We are totally addicted to it. I was even asked to extend its behaviour which forced me to write some custom Javascript. That's how popular it has become here.

Thats actully quite interesting as I wasn't awre of the history of the module. I wonder if the maintainer is still about? Theres a couple of auto opened issues on drupal.org and it would be good to get your extensions added. If the maintainer is not about should we ask if we can maintain it?

**Comment by andybroomfield** [comment]
Contrib modules for consideration from BHCC side:
[Tasty backend](https://www.drupal.org/project/tasty_backend) provides more sensible admin toolbar menu and seperated content dashboards at /admin alongside sensible default configuration to [menu_admin_per_menu](https://www.drupal.org/project/menu_admin_per_menu) and [role_delgation](role_delegation). We started experimenting with this for our blogs microsite before covid and whilst thats on hiatus the feedback was quite positive. 

I'd like to bring in Role delegation as that can fix issues with custom role handling on our current sites. So users don't need either the full manage user permissions to assign roles but also stop them assigning themselves a more senior role.

Honurable mention for [Flag](https://www.drupal.org/project/flag) becuase its a useful nice to have. Currently used to enable the alert banners accross the top and has been used for some internal dev work. May not make it in distribution though if uneeded.

**Comment by Adnan-cds** [comment]
+1 for role_delegation. menu_admin_per_menu looks useful too.

I have tried Tasty backend only once to become familiar with it.  The result looked very similar to what users with the "Communication" role already gets in our sites.  It was a very brief session, so I could be wrong.

**Comment by andybroomfield** [comment]
[Metatag](https://www.drupal.org/project/metatag) Is another that we should bring in and set default config for.

**Comment by andybroomfield** [comment]
@Adnan-cds Yes its very simmilar to our communications set up (spliting out the my dashboard content view), that and re-orginising the menu. We have some of this set up in custom bhcc_admin modules, though the more generic way tasty does it seems cleaner.

**Comment by stephen-cox** [comment]
I'm closing this as it's no longer relevant.
